### PR TITLE
Deprecate config_changes

### PR DESCRIPTION
**Pull request by TheNavigat**
Closes https://github.com/certbot/certbot/issues/7454

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
   staging server instead of the live server when `--dry-run` is used.
 * Updated certbot-dns-google to depend on newer versions of
   google-api-python-client and oauth2client.
+* Deprecated config_changes
```
Comment (bmw) [inline CHANGELOG.md]: nit: We've used "deprecate" to refer to features that still exist but will be removed in the future.

I'd essentially copy the changelog entries from https://github.com/certbot/certbot/pull/7198 but replace "has been deprecated and will be removed..." to "has been removed."

File: certbot/main.py [hunk 2]
```diff
@@ -977,9 +977,7 @@ def config_changes(config, unused_plugins):
     :rtype: None
 
     """
-    logger.warning("The config_changes subcommand has been deprecated"
-                   " and will be removed in a future release.")
-    client.view_config_changes(config)
+    logger.error("The config_changes subcommand has been deprecated.")
```
Comment (bmw) [inline certbot/main.py]: I think we should delete this function outright.

File: certbot/cli.py [hunk 3]
```diff
@@ -398,11 +398,6 @@ def _get_help_string(self, action):
...
         " --key-path /path/to/private-key [options]\n\n"
     }),
-    ("config_changes", {
-        "short": "Show changes that Certbot has made to server configurations",
-        "opts": "Options for viewing configuration changes",
-        "usage": "\n\n  certbot config_changes [options]\n\n"
-    }),
     ("rollback", {
         "short": "Roll back server conf changes made during certificate installation",
         "opts": "Options for rolling back server configuration changes",
```
Comment (bmw) [inline certbot/cli.py]: I can't comment on it directly, but I think we should also delete the line https://github.com/certbot/certbot/pull/7469/files#diff-98e51bbf59cb44db5dc06a7c5a002983R445.

File: certbot/tests/main_test.py [hunk 4]
```diff
@@ -753,15 +753,11 @@ def test_rollback(self, _):
...
             mock.ANY, 123, mock.ANY, mock.ANY)
 
-    def test_config_changes(self):
-        _, _, _, client = self._call(['config_changes'])
-        self.assertEqual(1, client.view_config_changes.call_count)
-
-    @mock.patch('certbot.main.logger.warning')
-    def test_config_changes_deprecation(self, mock_warning):
+    @mock.patch('certbot.main.logger.error')
+    def test_config_changes_deprecation(self, mock_error):
```
Comment (bmw) [inline certbot/tests/main_test.py]: If you make the change I suggested above, don't forget to delete this test.

File: CHANGELOG.md [hunk 5]
```diff
@@ -29,6 +29,7 @@ Certbot adheres to [Semantic Versioning](https://semver.org/).
 * acme.standalone.BaseRequestHandlerWithLogging and
   acme.standalone.simple_tls_sni_01_server have been deprecated and will be
   removed in a future release of the library.
+* Certbot's config_changes subcommand has been removed
```
Comment (bmw) [inline CHANGELOG.md]: Can you also note the functions that were removed like in https://github.com/certbot/certbot/pull/7198/files.

File: certbot/tests/cli_test.py [hunk 6]
```diff
@@ -186,10 +186,6 @@ def test_help(self):
         self.assertTrue("--delete-after-revoke" in out)
         self.assertTrue("--no-delete-after-revoke" in out)
 
-        out = self._help_output(['-h', 'config_changes'])
```
Comment (bmw) [inline certbot/tests/cli_test.py]: nit: The assertions that `--cert-path` and `--key-path` aren't included in the help output are mildly useful. Can we keep this code but change the subcommand used here from `config_changes` to something like `register`?

**Comment by TheNavigat** [comment]
> Also, it's not a big deal, but please do not rebase your PRs between reviews. Our request to do this is kind of buried in step 5 of [submitting a pull request](https://certbot.eff.org/docs/contributing.html#submitting-a-pull-request), but since we squash PRs when merging anyway, not rebasing allows the reviewer to easily track changes since their last review.

My bad. I'm still used to the old way of doing it where commits were not squashed, and back then rebasing was more preferred. Thanks for noting this out.

**Pull request merged**